### PR TITLE
model node

### PR DESCRIPTION
**Pull request by jckhang**
## What does this PR do?

move model related functions back to node. remove model blob

## Changes

- [x] Add make command to generated .d.ts for typescript and .pyi for python
- [x] Fix extending Custom Error 

> https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work
- [x] Move local pre-commit hook to github action
- [x] Move model related function back to NodeHandle

## Before reviewing

- [ ] Are there any linked PRs?
- [ ] Did you test this PR?
- [ ] Is there any migration or seeding necessary?
- [ ] Did you add api to only one of the language (python/typescript)?

### Links:

-   xyme-backend: https://github.com/Accern/xyme-backend/pull/141

### New API added that not implemented in other languages

If your answer is yes, please also update the todo_api.txt at where the API is missing.

### CODE REVIEW

File: packages/python/accern_xyme/accern_xyme.py [hunk 1]
```diff
@@ -661,16 +661,11 @@ def get_blob_type(self, blob_uri: str) -> BlobTypeResponse:
 
     def get_csv_blob(self, blob_uri: str) -> 'CSVBlobHandle':
         blob_type = self.get_blob_type(blob_uri)
+        print("blob_type", blob_type)
```
Comment (jckhang) [inline packages/python/accern_xyme/accern_xyme.py:664]: remove

File: packages/python/accern_xyme/accern_xyme.py [hunk 2]
```diff
@@ -1733,10 +1730,11 @@ def __init__(
         self._inputs: Dict[str, Tuple[str, str]] = {}
         self._state: Optional[int] = None
         self._config_error: Optional[str] = None
+        self._is_model: Optional[bool] = None
 
     def as_owner(self) -> BlobOwner:
         return cast(BlobOwner, {
```
Comment (jckhang) [inline packages/python/accern_xyme/accern_xyme.py:1736]: no cast

File: packages/python/accern_xyme/accern_xyme.py [hunk 3]
```diff
@@ -2387,14 +2413,14 @@ def add_from_file(
         finally:
             self._clear_upload()
 
-    def add_from_df(
+    def add_from_content(
```
Comment (jckhang) [inline packages/python/accern_xyme/accern_xyme.py:2416]: keep add_from_df

File: packages/python/accern_xyme/accern_xyme.py [hunk 4]
```diff
@@ -2387,14 +2413,14 @@ def add_from_file(
...
             self,
-            df: pd.DataFrame,
+            content: Any,
```
Comment (jckhang) [inline packages/python/accern_xyme/accern_xyme.py:2418]: content: Union[bytes, str, pd.DataFrame]

File: packages/python/accern_xyme/accern_xyme.py [hunk 5]
```diff
@@ -937,7 +932,9 @@ def read_dvc(
 
         res = dvc.api.read(path, repo=repo, rev=rev, mode="r")
         try:
-            return json_loads(res)
+            maybe_parse = maybe_json_loads(res)
+            if maybe_parse is not None:
+                return maybe_parse
```
Comment (jckhang) [inline packages/python/accern_xyme/accern_xyme.py:937]: no try catch

File: packages/python/accern_xyme/util.py [hunk 6]
```diff
@@ -127,10 +127,13 @@ def maybe_timestamp(timestamp: Optional[str]) -> Optional[pd.Timestamp]:
     return None if timestamp is None else pd.Timestamp(timestamp)
 
 
-def df_to_csv(df: pd.DataFrame) -> BytesIO:
+def content_to_csv_bytes(content: Any) -> BytesIO:
```
Comment (jckhang) [inline packages/python/accern_xyme/util.py:130]: keep df_to_csv
create another content_to_csv

File: packages/typescript/tsconfig-info.json [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+      "moduleResolution": "node",
+      "sourceMap": true,
+      "outDir": "../types",
+      "target": "es5",
+      "lib": [
+        "es2019"
+      ]
+    },
+    "include": ["index.ts"]
+  }
```
Comment (jckhang) [inline packages/typescript/tsconfig-info.json:16]: indent

File: requirements.txt [hunk 8]
```diff
@@ -1,7 +1,7 @@
 dvc>=2.0.17
 graphviz>=0.14.1
-numpy>=1.17.3
-pandas>=0.25.3
+numpy>=1.20.0
+pandas>=1.2.4
```
Comment (jckhang) [inline requirements.txt:4]: dump back versions, same in cfg

**Review by JosuaKrause — APPROVED** [review]
lgtm

**Pull request merged**